python/ray/train/data_parallel_trainer.py

```diff
@@ -450,13 +490,16 @@ def training_loop(self) -> None:
         # Shutdown workers.
         backend_executor.shutdown()
 
-    def get_dataset_config(self) -> Dict[str, DatasetConfig]:
+    def get_dataset_config(self) -> DataConfig:
         """Return a copy of this Trainer's final dataset configs.
 
         Returns:
             The merged default + user-supplied dataset config.
         """
-        return self._dataset_config.copy()
+        if isinstance(self._data_config, _LegacyDataConfigWrapper):
+            return self._data_config._dataset_config
```

Comment:
this is returning a `Dict[str, DatasetConfig]`, which doesn't match with the type hint.

Reply:
I think this is fine, to avoid polluting our type hints with deprecated fields.